datafusion/physical-expr-common/src/physical_expr.rs

```diff
@@ -193,33 +191,6 @@ pub fn with_new_children_if_necessary(
     }
 }
 
-/// Rewrites an expression according to new schema; i.e. changes the columns it
-/// refers to with the column at corresponding index in the new schema. Returns
-/// an error if the given schema has fewer columns than the original schema.
-/// Note that the resulting expression may not be valid if data types in the
-/// new schema is incompatible with expression nodes.
-pub fn with_new_schema(
```

Comment:
move to `physical_expr::expressions::column`

Reply:
Since this takes in a general `PhysicalExpr` and rewrites it, it is more natural for it to be a top-level utility. Do we have to nest it inside `expressions::column` due to some dependency issues? Avoiding that nesting would be good due to the signature/function of this routine.

Reply:
The best place we could move to is probably expressions::mod

Reply:
`expressions::mod` sounds good to me 👍